SUI/SUI/Views/ProductsView.swift

```diff
@@ -0,0 +1,188 @@
+        static let searchText = "Search..."
+        static let filterImageName = "filters"
+    }
+
+    // MARK: - Public Properties
+
+    var body: some View {
+        NavigationView {
+            VStack {
+                ZStack {
+                    topGradientView
+                }
+                totalCostTextView
+                productsView
+            }
+        }
+    }
+
+    // MARK: - Private Properties
+
+    @StateObject private var viewModel = ProductsViewModel()
+
+    private var productsView: some View {
+        ScrollView {
+            VStack {
+                ForEach(viewModel.products, id: \.id) { product in
+                    makeProductView(product)
+                }
+            }
+        }
+    }
+
+    private var totalCostTextView: some View {
+        Text("\(Constants.totalCost) \(viewModel.totalSum)")
+            .font(.headline)
+            .bold()
+    }
+
+    private var topGradientView: some View {
+        VStack {
+            EmptyView()
+            searchBarView
+        }
+        .frame(
+            width: UIScreen.main.bounds.width,
+            height: 80
+        )
+        .background(
+            LinearGradient(
+                colors: [.yellow, .orange],
+                startPoint: .leading,
+                endPoint: .trailing
+            )
+        )
+        .ignoresSafeArea()
+    }
+
+    private var searchBarView: some View {
+        ZStack {
+            RoundedRectangle(cornerRadius: 20)
+                .frame(
+                    width: 250,
+                    height: 40
+                )
+                .foregroundColor(.white)
+            Image(systemName: Constants.glassImageName)
+                .resizable()
+                .frame(
+                    width: 25,
+                    height: 25
+                )
+                .foregroundColor(.gray)
+                .padding(.leading, -115)
+            Text(Constants.searchText)
+                .font(.body)
+                .bold()
+                .foregroundColor(.purple)
+                .padding(.leading, -80)
+            NavigationLink(
+                isActive: $viewModel.isFilterButtonTapped,
+                destination: {
+                    FiltersView()
+                        .navigationBarBackButtonHidden(true)
+                },
+                label: {
+                    Button {
+                        viewModel.isFilterButtonTapped = true
+                    } label: {
+                        Image(Constants.filterImageName)
+                            .resizable()
+                            .frame(
+                                width: 30,
+                                height: 30
+                            )
+                            .foregroundColor(.white)
+                            .padding(.leading, 300)
+                    }
+                }
+            )
+        }
+    }
+
+    private func makeProductView(_ model: Product) -> some View {
```

Comment:
```suggestion
    private func makeProductView(_ model: Product) -> some View {
```
Нужно разнести makeProductView на более мелкие вью